### ISSUE TITLE
Recovery doesn't support self-hosted etcd without TLS

### ISSUE THREAD

**Issue by bison** [kubernetes-incubator/bootkube#612]
It looks like after #592, `bootkube recover` doesn't work with a self-hosted etcd that doesn't use TLS. I get errors about required assets being missing:

```sh-session
# GLOG_v=4 /home/core/bootkube recover --recovery-dir=/root/recover-2017-06-24 --etcd-backup-file=/root/etcd-backup-2017-06-24 --kubeconfig=/etc/kubernetes/kubeconfig
Attempting recovery using etcd backup file at "/root/etcd-backup-2017-06-24"...
Writing asset: /etc/kubernetes/manifests/recovery-etcd.yaml
Waiting for etcd server to start...
I0624 20:14:52.195192   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:15:02.195125   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:15:12.195214   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:15:22.195115   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:15:32.195299   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:15:42.195126   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:15:52.195108   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:16:02.196252   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:16:12.195179   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:16:22.195130   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:16:32.195156   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:16:42.197720   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:16:52.195221   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:17:02.195295   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:17:12.195125   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:17:22.195116   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:17:32.195199   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
I0624 20:17:42.195130   17731 util.go:25] could not create etcd client, retrying later: grpc: timed out when dialing
Error: failed to extract some required objects: [etcd-member-peer-tls etcd-member-client-tls etcd-operator-client-tls]
failed to extract some required objects: [etcd-member-peer-tls etcd-member-client-tls etcd-operator-client-tls]
```

Recovery of the assets works fine with a version built before that PR. Is that something that's going to be supported going forward?

**Comment by hongchaodeng** [comment]
@aaronlevy 
Does bootkube want to handle non-TLS case in the future?

**Comment by aaronlevy** [comment]
Hmm, my initial thought is, no -- mostly because we won't allow using self-hosted etcd *without* tls from this point forward (I don't really see a case for allowing it).

So the recovery tool wouldn't be able to support clusters launched prior to tls being added -- which is unfortunate (but we're still in an alpha phase for both recovery, and self-hosted etcd).

**Comment by klausenbusk** [comment]
Is recovery even working for self-hosted etcd with TLS?
`bootkube recover --recovery-dir=/home/core/recovered --etcd-backup-file=/home/core/etcdbackup --kubeconfig=/etc/kubernetes/kubeconfig` works and generate the assets, but `bootstrap-etcd` can't be started, for some reason (still debugging)...

```
Jul 16 16:38:56 k8s-m1 kubelet-wrapper[29979]: E0716 16:38:56.775185   29979 pod_workers.go:182] Error syncing pod f89c4f31133c2047638b9c4a28de0c16 ("bootstrap-etcd-165.227.xx.xx_kube-system(f89c4f31133c2047638b9c4a28de0c16)"), skipping: timeout expired waiting for volumes to attach/mount for pod "kube-system"/"bootstrap-etcd-165.227.xx.xx". list of unattached/unmounted volumes=[etcd etcdbackup secrets]
```

**Comment by klausenbusk** [comment]
> but bootstrap-etcd can't be started, for some reason (still debugging)...

This was caused by some old `bootstrap` containers, due to another bug*.
So:
```
sudo sysetmctl stop kubelet
docker rm -f $(docker ps -aq)
sudo systemctl start kubelet
sudo /home/core/bootkube start --asset-dir=/home/core/recovered
```

\* `etcd-0000` can't start if `/var/etcd/` contain some old etcd data. So people also need to log into every master node and empty the `/var/etcd` dir..

**Comment by xiang90** [comment]
@klausenbusk 

Both docker state, active manifest and etcd state need to be cleaned up. Kubelet needs to be restarted.

We should add a bootkube recovery cleanup or add cleanup process into the bootkube recovery sub-command. 

If you have time, can you submit a PR to improve it?

**Comment by aaronlevy** [comment]
opened https://github.com/kubernetes-incubator/bootkube/issues/651 to track the above

**Comment by dghubble** [comment]
I'd think this is a non-goal. Do we want to allow bootkube clusters with non-TLS etcd setups to begin with?

**Comment by ericchiang** [comment]
Self-hosted etcd has been removed from bootkube https://github.com/kubernetes-incubator/bootkube/issues/738

Closing this issue for now. If/when we decide to re-enable it we can re-open.
